docs/en/styling-and-theming/basic-usage.md

```diff
@@ -0,0 +1,177 @@
+import ThemedMixin, { theme } from '@dojo/framework/widget-core/mixins/Themed';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+import * as css from '../styles/MyWidget.m.css';
+
+@theme(css)
+export default class MyWidget extends ThemedMixin(WidgetBase) {
+	protected render() {
+		return <div classes={[this.theme(css.root)]}>My Widget</div>;
+	}
+}
+```
+
+## Creating a theme
+
+Overriding a default CSS class that a widget uses with [custom theme properties](./supplemental.md#working-with-themes):
+
+> src/themes/MyTheme/MyWidget.m.css
+
+```css
+.root {
+	color: hotpink;
+	background-color: slategray;
+}
+```
+
+> src/themes/MyTheme/theme.ts
+
+```ts
+import * as myWidgetCss from './MyWidget.m.css';
+
+export default {
+	'my-app/MyWidget': myWidgetCss
+};
+```
+
+## Abstracting common theme properties
+
+Making use of [CSS custom properties](./supplemental.md#css-custom-properties):
+
+> src/themes/variables.css
+
+```css
+:root {
+	--foreground: hotpink;
+	--background: slategray;
+}
+```
+
+> src/themes/MyTheme/MyWidget.m.css
+
+```css
+@import '../variables.css';
+
+.root {
+	color: var(--foreground);
+	background-color: var(--background);
+}
+```
+
+## Specifying a default application theme
+
+Using the [theme injector function](./supplemental.md#making-themeable-applications) to automatically provide a custom theme to all themeable widgets in an application:
+
+> src/main.tsx
+
+```tsx
+import renderer from '@dojo/framework/widget-core/vdom';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Registry from '@dojo/framework/widget-core/Registry';
+import { registerThemeInjector } from '@dojo/framework/widget-core/mixins/Themed';
+
+import myTheme from './themes/MyTheme/theme';
+import App from './widgets/App';
+
+const registry = new Registry();
+registerThemeInjector(myTheme, registry);
+
+const r = renderer(() => <App />);
+r.mount({ registry });
+```
+
+## Changing the theme within an application
+
+Using the [ThemeSwitcher utility widget](./supplemental.md#changing-the-currently-active-theme) to allow users to choose between available themes, and enact a theme change via [`ThemeSwitcher`'s `updateTheme` method](./supplemental.md#themeswitcher-properties). This should be used together with [`registerThemeInjector`](./supplemental.md#making-themeable-applications) to reactively propagate theme changes to all themeable widgets.
+
+> src/widgets/App.tsx
+
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import ThemedMixin, { ThemeSwitcher, theme, UpdateTheme } from '@dojo/framework/widget-core/mixins/Themed';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+import myTheme from '../themes/MyTheme/theme';
+import alternativeTheme from '../themes/MyAlternativeTheme/theme';
+
+import * as css from '../styles/App.m.css';
+
+@theme(css)
+class App extends ThemedMixin(WidgetBase) {
+	protected render() {
+		return (
+			<div classes={[this.theme(css.container)]}>
```

Comment:
perhaps drop the wrapping divs to make this a bit smaller, the formatting is a little off as well. could just be GHs big spaces though

Reply:
This is all prettier auto-formatted at the moment!